dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -327,30 +328,32 @@ private int getCountContentletsReferencingStep(WorkflowStep step) throws DotData
 	}
 
 	@WrapInTransaction
-	public void reorderStep(WorkflowStep step, int order) throws DotDataException, AlreadyExistException {
-		WorkflowScheme scheme = findScheme(step.getSchemeId());
-		List<WorkflowStep> steps = null;
+	@Override
+	public void reorderStep(final WorkflowStep step, final int order) throws DotDataException, AlreadyExistException {
+
+		final List<WorkflowStep> steps;
+		final WorkflowScheme scheme = this.findScheme(step.getSchemeId());
 
 		try {
-			steps = findSteps(scheme);
+
+			steps  = this.findSteps (scheme);
 		} catch (Exception e) {
-			throw new DotDataException(e.getLocalizedMessage());
-		}
-		List<WorkflowStep> newSteps = new ArrayList<WorkflowStep>();
-		order = (order < 0) ? 0 : (order >= steps.size()) ? (steps.size() - 1) : order;
-		for (int i = 0; i < steps.size(); i++) {
-			WorkflowStep s = steps.get(i);
-			if (s.equals(step)) {
-				continue;
-			}
-			newSteps.add(s);
+			throw new DotDataException(e.getLocalizedMessage(), e);
 		}
 
-		newSteps.add(order, step);
-		int newOrder=0;
-		for(WorkflowStep newStep : newSteps){
-			newStep.setMyOrder(newOrder++);
-			saveStep(newStep);
+		IntStream.range(0, steps.size())
+				.filter(i -> steps.get(i).getId().equals(step.getId()))

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'i' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)